src/evaluators/Program.js

```diff
@@ -222,11 +224,21 @@ export default function(ast: BabelNodeProgram, strictCode: boolean, env: Lexical
 
   GlobalDeclarationInstantiation(realm, ast, env, strictCode);
 
+  let context = realm.getRunningContext();
   let val;
 
   for (let node of ast.body) {
     if (node.type !== "FunctionDeclaration") {
-      let potentialVal = env.evaluate(node, strictCode);
+      let potentialVal = env.evaluateCompletion(node, strictCode);
+      if (potentialVal instanceof Completion) {
+        if (!realm.useAbstractInterpretation) throw potentialVal;
+        // todo: emit code to throw exeptions at runtime
+        throw new FatalError();
```

Comment:
What's the user experience when this FatalError is hit? As no other error gets reported, I'd expect at least some verbal message indicating what limitation the user hit.

Reply:
The todo bit is the near future. For now, there just be a generic message about prepack failing.